src/shared/local-addresses.c

```diff
@@ -142,11 +142,46 @@ int local_addresses(sd_netlink *context, int ifindex, int af, struct local_addre
         return (int) n_list;
 }
 
+static int add_local_gateway(
+                struct local_address **list,
+                size_t *n_list,
+                size_t *n_allocated,
+                int af,
+                int ifindex,
+                uint32_t metric,
+                const RouteVia *via) {
+
+        struct local_address *a;
+
+        assert(list);
+        assert(n_list);
+        assert(n_allocated);
+        assert(via);
+
+        if (af != AF_UNSPEC && af != via->family)
+                return 0;
+
+        if (!GREEDY_REALLOC(*list, *n_allocated, *n_list + 1))
+                return -ENOMEM;
+
+        a = *list + *n_list;
+
+        *a = (struct local_address) {
+                .ifindex = ifindex,
+                .metric = metric,
+                .family = via->family,
+                .address = via->address,
+        };
+
+        (*n_list)++;
```

Comment:
maybe:

```c

`(*list)[(*n_list)++] = (struct local_address) {…
```

i.e. merge the las three lines in one? After all conceptually it's one operation: append an entry to the array...